Packs/QualysVulnerabilityManagement/Integrations/QualysVulnerabilityManagement/QualysVulnerabilityManagement.py

```diff
@@ -0,0 +1,931 @@
+    'qualys-virtual-host-manage': {'context_prefix': 'Qualys.VirtualEndpoint', 'context_key': 'DATETIME',
+                                   'args': ['action', 'ip', 'network_id', 'port', 'fqdn'],
+                                   'api_route': API_SUFFIX + 'asset/vhost/', 'call_method': 'POST', 'resp_type': 'text',
+                                   'json_path': "SIMPLE_RETURN.RESPONSE", 'table_name': ''},
+    'test-module': {'api_route': API_SUFFIX + '/scan/?action=list', 'call_method': 'POST', 'resp_type': 'text'},
+    'args_values': {}, 'inner_args_values': {}
+}
+
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+
+    def __init__(self, base_url, username, password, verify=True, proxy=False,
+                 ok_codes=tuple(), headers=None):
+        super().__init__(base_url, verify, proxy, ok_codes, headers, auth=(username, password))
+
+    def command_http_request(self, command_data: Dict) -> requests.Response:
+        """
+        Make a http request to Qualys API
+        Args:
+            command_data: Dictionary with information about the command requested
+        Returns:
+            response from Qualys API
+        Raises:
+            DemistoException: can be raised by the _http_requeset function
+        """
+        return self._http_request(
+            method=command_data['call_method'],
+            url_suffix=command_data['api_route'],
+            params=commands_data['args_values'],
+            resp_type=command_data['resp_type'],
+            timeout=60)
+
+
+''' HELPER FUNCTIONS '''
+
+
+def json_get_path(json_obj: Dict[str, Any], path_str: str) -> Dict[str, Any]:
+    """
+    This function takes a path and gets the object at the specified path in the dictionary
+    Args:
+        json_obj: Dictionary
+        path_str: String of nested keys '.' separated, making a path
+    Returns:
+            Dict: dictionary nested inside the original dictionary or the original dict
+    Raises:
+        KeyError: raised if a bad or incorrect path was provided or bad json object
+        TypeError: raised if json_obj is not a dictionary
+    """
+    if len(path_str) == 0:
+        return json_obj
+    path_keys_list_path = path_str.split(".")
+    current_json_obj = json_obj
+    for key in path_keys_list_path:
+        current_json_obj = current_json_obj[key]
+    return current_json_obj
+
+
+def create_ip_list_dict(res_json: Dict[str, Any], type_of_dict: str) -> Dict[str, Any]:
+    """
+    Creates a dictionary of a range type of ips or single address type of ips
+    Args:
+        res_json: Dictionary received from ip list command with 'Address' or 'Range' keys
+        type_of_dict: The wanted type of dictionary: 'Range' or 'Address'
+    Returns:
+        A dictionary with the specified type of ips values
+    """
+    ips_dict = {}
+
+    if type_of_dict in res_json:
+        ips = res_json[type_of_dict]
+        if isinstance(ips, str) or isinstance(ips, Dict):
+            ips_dict = {'0': ips}
```

Comment:
When there's a single IP address or a single range, Qualys doesn't return the usual list of objects, as can be seen in the else case. 

When there's a single value it returns either a single string or a single Dict so I couldn't iterate over it. Therefore I created a condition for this case

Reply:
Got it, so in that case - add a developer comment there that will explain the reason for it.
It would be really hard to understand it in the future without the comment.

Reply:
Added.